### DISCUSSION TITLE
Updating Settings Stored in the (IndexDB) ItemTable

### DISCUSSION THREAD

**Discussion by barrycarr** [gitpod-io/openvscode-server#423] [Q&A] [answered]
Hi

Further to my previous question, #421, the settings I need to change the defaults for are stored in the workspace's ItemTable; they're not user settings. I've tried setting the values I need to change using the technique suggested in the answer to #421 but, as the setting I trying to change isn't a "public" setting, ie: something that would be found in a users `settings.json` , adding my new config values to `configurationDefaults`  doesn't work  as the config keys I'm trying to set aren't  registered in the workspace's instance of `ConfigurationRegistry`. I know this because I've stepped through the configuration process and my new values are being ignored. 

So, my question is this: is there a way to set configuration values that are usually stored in the workspace's ItemTable? 

To give you a concrete value, I need to change this key `workbench.view.debug.state` so that some of the builtin debugger features are hidden and our custom panels are visible and not collapsed. Is this possible by any means? I'm open to (and grateful for) any suggestions that will enable me to do this. 

Thanks very much.
Cheers

**Comment by jeanp413** [comment, marked as answer]
If the setting you want to change is not in `settings.json` then there's no way to change it from the default value as far as I know, your only option is to change it in the vscode source code itself
